src/util/GuardedStream.ts

```diff
@@ -29,47 +27,30 @@ export function isGuarded<T extends NodeJS.EventEmitter>(stream: T): stream is G
 }
 
 /**
- * Makes sure that listeners always receive the error event of a stream,
- * even if it was thrown before the listener was attached.
- * If the input is already guarded nothing will happen.
- * @param stream - Stream that can potentially throw an error.
- *
- * @returns The stream.
- */
-export function guardStream<T extends NodeJS.EventEmitter>(stream: T): Guarded<T> {
-  const guarded = stream as Guarded<T>;
-  if (!isGuarded(stream)) {
-    guarded[guardedErrors] = [];
-    attachDefaultErrorListener.call(guarded, 'error');
-  }
-  return guarded;
-}
-
-/**
- * Callback that is used when a stream emits an error and no error listener is attached.
+ * Callback that is used when a stream emits an error and no other error listener is attached.
  * Used to store the error and start the logger timer.
+ *
+ * It is important that this listener always remains attached for edge cases where an error listener gets removed
+ * and the amount of error listeners is checked immediately afterwards.
```

Comment:
```suggestion
 * and the number of error listeners is checked immediately afterwards.
```

But we can also more safely write: it's safer to keep it attached to avoid edge cases.